load_tests/boa_locust.py

```diff
@@ -0,0 +1,221 @@
+"""
+BOA load test script for use with the Locust tool (https://locust.io).
+
+Usage:
+
+From the top-level Teena directory, install dependencies.
+pip -r requirements.txt
+
+Start Locust with this file.
+locust -f load_tests/boa_locust.py
+
+In your browser, go to localhost:8089 to start tests and view charts.
+
+Visit https://locust.io for more information.
+"""
+
+
+from itertools import groupby
+import os
+import random
+import yaml
+
+from locust import HttpLocust, TaskSet, task
+import psycopg2
+import psycopg2.extras
+from pyquery import PyQuery
+
+
+"""
+Define utility functions.
+"""
+
+
+def merge_recursive(target, source):
+    for key in source:
+        if key in target and isinstance(target[key], dict) and isinstance(source[key], dict):
+            merge_recursive(target[key], source[key])
+        else:
+            target[key] = source[key]
+
+
+def sample(_list):
+    return _list[random.randint(0, len(_list) - 1)]
+
+
+"""
+Set configurations.
+"""
+
+
+with open('settings.yml') as config_yml:
+    configs = yaml.safe_load(config_yml)
+with open(os.environ.get('HOME') + '/.webdriver-config/settings.yml') as local_config_yml:
+    local_configs = yaml.safe_load(local_config_yml)
+merge_recursive(configs, local_configs)
+
+
+boac_cfg = configs['boac']
+nessie_cfg = configs['nessie']
+
+
+"""
+Define database connections.
+"""
+
+
+class Rds():
+    @classmethod
+    def fetch(cls, sql, params=None):
+        connection = psycopg2.connect(**cls.connection_params)
+        cursor = connection.cursor(cursor_factory=psycopg2.extras.DictCursor)
+        cursor.execute(sql, params)
+        results = [dict(row) for row in cursor.fetchall()]
+        cursor.close()
+        connection.close()
+        return results
+
+
+class BoaRds(Rds):
+    connection_params = {
+        'dbname': boac_cfg['db_name'],
+        'user': boac_cfg['db_user'],
+        'password': boac_cfg['db_password'],
+        'host': boac_cfg['db_host'],
+        'port': boac_cfg['db_port'],
+    }
+
+
+class NessieRds(Rds):
+    connection_params = {
+        'dbname': nessie_cfg['pg_db_name'],
+        'user': nessie_cfg['pg_db_user'],
+        'password': nessie_cfg['pg_db_password'],
+        'host': nessie_cfg['pg_db_host'],
+        'port': nessie_cfg['pg_db_port'],
+    }
+
+
+"""
+Load test data.
+"""
+
+
+def _row_to_cohort(row):
+    return {
+        'id': row['cohort_id'],
+        'student_count': row['student_count'],
+        'filter_criteria': row['filter_criteria'],
```

Comment:
@paulkerschen – Intentionally `snake_case`?

Reply:
@johncrossman Yes, Teena's house style stays pretty snaky in cases this like.